src/LocalCoverage.jl

```diff
@@ -4,199 +4,246 @@ using CoverageTools
-    joinpath(dirname(Base.locate_package(Base.PkgId(pkgstr))), "..")
+Base.@kwdef struct FileCoverageSummary
+    "File path relative to the directory of the project"
+    filename::String
+    "Number of lines covered by tests"
+    lines_hit::Int
+    "Number of lines with content to be tested"
+    lines_tracked::Int
+    "Percentage of lines covered"
+    coverage::Float64
+    "List of all line ranges without coverage"
+    coverage_gaps::Vector{UnitRange{Int}}
 end
-pkgdir(m::Module) = joinpath(dirname(pathof(m)), "..")
 
 """
-$(SIGNATURES)
+Summarized coverage data about a specific package. Contains a list of
+`FileCoverageSummary` relative to the package files as well as global metrics
+about the package coverage.
 
-Open the HTML coverage results in a browser for `pkg` if they exist.
+See [`report_coverage`](@ref).
 
-See [`generate_coverage`](@ref).
+$(FIELDS)
 """
-function open_coverage(pkg;
-                       coverage_file::AbstractString=joinpath(COVDIR, "index.html"))
-    htmlfile = joinpath(pkgdir(pkg), coverage_file)
-    if !isfile(htmlfile)
-        @warn("Not found, run generate_coverage(pkg) first.")
-        return nothing
-    end
-    try
-        if Sys.isapple()
-            run(`open $htmlfile`)
-        elseif Sys.islinux() || Sys.isbsd()
-            run(`xdg-open $htmlfile`)
-        elseif Sys.iswindows()
-            run(`start $htmlfile`)
-        end
-    catch e
-        error("Failed to open the generated $(htmlfile)\n",
-              "Error: ", sprint(Base.showerror, e))
-    end
-    nothing
+Base.@kwdef struct PackageCoverage
+    "Absolute path of the package"
+    package_dir::String
+    "List of files coverage summaries tracked"
+    files::Vector{FileCoverageSummary}
+    "Total number of lines covered by tests in the package"
+    lines_hit::Int
+    "Total number of lines with content to be tested in the package"
+    lines_tracked::Int
+    "Percentage of package lines covered"
+    coverage::Float64
 end
 
 """
 $(SIGNATURES)
 
-Returns a table giving coverage details by file in human readable form:
-- column 1: source file name
-- column 2: tuple (lines_hit, lines_tracked)
-- column 3: coverage fraction
+Get the root directory of a package.
 """
-function coverage_summary_table(coverage)
-    n = length(coverage)
-    tab = Array{Any, 2}(undef, 1+n, 3)
-    total_hit = 0
-    total_tracked    = 0
+pkgdir(pkgstr::AbstractString) = abspath(joinpath(dirname(Base.find_package(pkgstr)), ".."))
 
-    for (i, f) in enumerate(coverage)
-        hit     = count(x->!isnothing(x) && x>0, f.coverage)
-        tracked = count(x->!isnothing(x),        f.coverage)
-
-        total_hit     += hit
-        total_tracked += tracked
+"""
+$(SIGNATURES)
 
-        tab[i,:] .= (f.filename, (hit, tracked),
-                     100 * hit / tracked)
+Evaluate the ranges of lines without coverage.
+"""
+function find_gaps(coverage)
+    i = 1
+    last_line = length(coverage)
```

Comment:
```suggestion
    i, last_line = firstindex(coverage), lastindex(coverage)
```
(generalized indexing)

Reply:
Nice one